### PR TITLE
Support OpenVX delegate for several NPU/GPU

### PR DESCRIPTION
**Pull request by ayakoakasaka**
Add VX delegation as an external delegation of TFLite. By using it, several NPU/GPU (from VeriSilicon, NXP, Amlogic) can be controlled via WASI-NN.  Test Code can work with X86 simulator

### CODE REVIEW

File: core/iwasm/libraries/wasi-nn/README.md [hunk 1]
```diff
@@ -71,6 +72,18 @@ docker run \
...
 ```
 
+* vx-delegate for NPU (x86 simulater)
+
+```
+docker run \
+    -v $PWD/core/iwasm/libraries/wasi-nn/test:/assets wasi-nn-vx-delegate \
+    --dir=/assets \
+    --env="TARGET=gpu" \
+    /assets/test_tensorflow.wasm
```
Comment (tonibofarull) [inline core/iwasm/libraries/wasi-nn/README.md:82]: is the test running in vx-delegate with `--env="TARGET=gpu"`?

File: core/iwasm/libraries/wasi-nn/src/wasi_nn_tensorflowlite.cpp [hunk 2]
```diff
@@ -206,6 +210,17 @@ tensorflowlite_init_execution_context(void *tflite_ctx, graph g,
                 NN_ERR_PRINTF("Error when enabling GPU delegate.");
                 use_default = true;
             }
+#elif defined(WASI_NN_ENABLE_EXTERNAL_DELEGATE)
+            NN_WARN_PRINTF("external delegation enabled.");
+            auto options = TfLiteExternalDelegateOptionsDefault(
+                "/usr/lib/libvx_delegate.so");
```
Comment (tonibofarull) [inline core/iwasm/libraries/wasi-nn/src/wasi_nn_tensorflowlite.cpp]: Can we make this parameterizable? Other devices will also use the external delegate functionality.

File: core/iwasm/libraries/wasi-nn/README.md (hunk 1 above)
Comment (ayakoakasaka) [reply core/iwasm/libraries/wasi-nn/README.md:82]: So far, yes. This is the question where the external delegation should be categorised

File: core/iwasm/libraries/wasi-nn/test/Dockerfile.vx-delegate [hunk 3]
```diff
@@ -86,8 +86,10 @@ WORKDIR /home/wamr/core/iwasm/libraries/wasi-nn/test/build
 
 RUN cmake \
     -DCMAKE_LIBRARY_PATH=/usr/local/lib/ \
+    -DCMAKE_INCLUDE_PATH=/usr/local/include/ \
     -DWAMR_BUILD_WASI_NN=1 \
     -DWAMR_BUILD_WASI_NN_ENABLE_EXT=1 \
+    -DWASI_NN_EXT_DELEGATE_PATH="/usr/lib/libvx_delegate.so" \
```
Comment (ayakoakasaka) [inline core/iwasm/libraries/wasi-nn/test/Dockerfile.vx-delegate:92]: I added the external delegate library path from the Config

File: core/iwasm/libraries/wasi-nn/src/wasi_nn_tensorflowlite.cpp (hunk 2 above)
Comment (ayakoakasaka) [reply core/iwasm/libraries/wasi-nn/src/wasi_nn_tensorflowlite.cpp]: Add compile option as below

**Review by tonibofarull — APPROVED** [review]

**Comment by ayakoakasaka** [comment]
Thank you @tonibofarull, I create same PR to upstream

**Pull request closed without merging**